### PR TITLE
Selinux proposal

### PR DESCRIPTION
**Pull request by jreidinger**
trello: https://trello.com/c/g94xQBDM/2159-8-musthave-selinux-and-security-refac

related: https://github.com/yast/yast-security/pull/83

### CODE REVIEW

File: src/lib/y2firewall/proposal_settings.rb [hunk 1]
```diff
@@ -59,6 +59,7 @@ def initialize
       # chose a different one in the proposal
       @default_zone = "public"
       @selinux_config = Y2Security::SelinuxConfig.new
+      @selinux_config.save # lets write the proposal to sync the initial state
```
Comment (dgdavid) [inline src/lib/y2firewall/proposal_settings.rb:62]: But current implementation of [SelinuxConfig#save](https://github.com/yast/yast-security/blob/2ccd5c23b7bc1f6fd77d6b9e8ff55d199a1b6258/src/lib/y2security/selinux_config.rb#L128-L148) will not write the initial state unless [SelinuxConfig#configurable?](https://github.com/yast/yast-security/blob/2ccd5c23b7bc1f6fd77d6b9e8ff55d199a1b6258/src/lib/y2security/selinux_config.rb#L150-L159)

Is that ok?

Comment (jreidinger) [reply src/lib/y2firewall/proposal_settings.rb:62]: yes, it is exactly what I need. As if it is not configurable, we do not touch it, so it is basically nothing and we do not display anything, neither write.

File: package/yast2-firewall.spec [hunk 2]
```diff
@@ -26,8 +26,10 @@ Url:            https://github.com/yast/yast-firewall
 
 Source0:        %{name}-%{version}.tar.bz2
 
-BuildRequires:  perl-XML-Writer update-desktop-files yast2-testsuite
+BuildRequires:  update-desktop-files
 BuildRequires:  yast2-devtools >= 4.2.2
+# for proposing selinux
+BuildRequires:  yast2-security >= 4.2.15
```
Comment (dgdavid) [inline package/yast2-firewall.spec:32]: So, we should merge https://github.com/yast/yast-security/pull/83, right?

File: src/lib/y2firewall/dialogs/proposal.rb [hunk 3]
```diff
@@ -55,6 +55,21 @@ def contents
             )
           )
         )
+
+        if @settings.selinux_config.configurable?
+          res.params << Frame(
```
Comment (dgdavid) [inline src/lib/y2firewall/dialogs/proposal.rb:60]: Curious. 

NP: I would have used a `#selinux_content` method returning `Empty()` when not configurable and called it inside the above `VBox`. Much readable to me, but maybe it does not work.

**Review by dgdavid — APPROVED** [review]
LGTM, just added some comments but it looks good anyway. Thanks!

File: package/yast2-firewall.spec (hunk 2 above)
Comment (jreidinger) [reply package/yast2-firewall.spec:32]: yes, it is dependency, otherwise test cannot be run

File: src/lib/y2firewall/dialogs/proposal.rb (hunk 3 above)
Comment (jreidinger) [reply src/lib/y2firewall/dialogs/proposal.rb:60]: I worry that Empty can break a bit layout as Empty is empty space on screen and no noop. That is why I dynamicaly add that frame.

**Comment by jreidinger** [comment]
It is still failing as in OBS is not the new version of security. It probably needs to be submitted together there?

**Comment by dgdavid** [comment]
> It is still failing as in OBS is not the new version of security. It probably needs to be submitted together there?

I submitted it with fixed dependencies two hours ago. Not sure when will be available in the container.

**Comment by jreidinger** [comment]
@lslezak when playing with PackagesProposal for pattern I get this failure when pattern is not available:

selinux_failure2

I think that issue is caused by fact that statuses does not find anything as such resolvable is not found?

**Comment by jreidinger** [comment]
OK, so for layout I tried HVCenter approach and Stretch one.

HVCenter does not look like I wanted:
selinux_hvcenter
and here is Stretch one ( final one )
selinux_stretches

File: package/yast2-firewall.changes [hunk 4]
```diff
@@ -1,8 +1,15 @@
+-------------------------------------------------------------------
+Thu Feb  4 09:14:13 UTC 2021 - Josef Reidinger <jreidinger@suse.com>
+
+- Add to firewall/security proposal option to setup selinux if
+  given product require it. (jsc#SLE-17427)
```
Comment (dgdavid) [inline package/yast2-firewall.changes:5]: NP: I'm using jsc#SMO-20, and jsc#SLE-17342. The latest is the DEV task.

File: package/yast2-firewall.spec [hunk 5]
```diff
@@ -26,8 +26,10 @@ Url:            https://github.com/yast/yast-firewall
 
 Source0:        %{name}-%{version}.tar.bz2
 
-BuildRequires:  perl-XML-Writer update-desktop-files yast2-testsuite
+BuildRequires:  update-desktop-files
 BuildRequires:  yast2-devtools >= 4.2.2
+# for proposing selinux
+BuildRequires:  yast2-security >= 4.2.16
```
Comment (dgdavid) [inline package/yast2-firewall.spec:32]: 4.2.17 includes file exceptions handlers and returns no patterns when SELinux mode is disabled.

File: src/lib/y2firewall/widgets/proposal.rb [hunk 6]
```diff
@@ -200,5 +200,30 @@ def help
...
+    # Widget to set SELinux mode
+    class SelinuxMode < CWM::ComboBox
+      def initialize(settings)
+        textdomain "firewall"
+
+        @settings = settings
+      end
+
+      def label
+        _("SELinux Mode")
```
Comment (dgdavid) [inline src/lib/y2firewall/widgets/proposal.rb:213]: As said, maybe in the context of the proposal  "SELinux" here is a bit redundant.

**Review by dgdavid — APPROVED** [review]
LGTM, just np/minor comments

File: src/lib/y2firewall/widgets/proposal.rb (hunk 6 above)
Comment (jreidinger) [reply src/lib/y2firewall/widgets/proposal.rb:213]: yeah, we also discuss it with @kobliha my argument is that SELinux Mode is easily googable, on other hand selinux is already in frame. So looks like you see it also duplicite, so lets remove it.

Comment (dgdavid) [reply src/lib/y2firewall/widgets/proposal.rb:213]: > SELinux Mode is easily googable

Should we add some useful help to the dialog?

File: package/yast2-firewall.changes (hunk 4 above)
Comment (jreidinger) [reply package/yast2-firewall.changes:5]: too many jira entries. I am using epic mentioned in dev task

File: src/lib/y2firewall/widgets/proposal.rb [hunk 7]
```diff
@@ -200,5 +200,43 @@ def help
...
+        self.value = @settings.selinux_config.mode.id
+      end
+
+      def store
+        @settings.selinux_config.mode = value
+      end
+
+      def help
+        _(
+          "<p>Sets default selinux mode. Modes are: <ul>" \
```
Comment (dgdavid) [inline src/lib/y2firewall/widgets/proposal.rb:231]:
```suggestion
          "<p>Sets default SELinux mode. Modes are: <ul>" \
```

**Review by dgdavid — APPROVED** [review]
Still looking good :) Just a minor suggestion that, of course, you can dismiss :)

Thanks a lot!

**Pull request merged**